source/lac/petsc_parallel_vector.cc

```diff
@@ -85,6 +85,20 @@ namespace PETScWrappers
 
 
 
+    Vector::Vector(const Vector &v)
+      : VectorBase()
+      , communicator(v.communicator)
+    {
+      if (v.has_ghost_elements())
+        Vector::create_vector(v.size(), v.local_size(), v.ghost_indices);
+      else
+        Vector::create_vector(v.size(), v.local_size());
```

Comment:
This is the right solution. The `=default` copy constructor worked because the `Vec` objects we store after getting it from PETSc is a pointer. But that doesn't do what we want: Once either the original or the copied `PETScWrappers::Vector` object goes out of scope, the pointer is deleted and the other object is left in a dangling state. Also, if one vector changes the size, the other one has inconsistent internal data structures. It really requires deep copying the object.